src/components/Wallet.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <div class="wallet">
+    <template v-if="!storedKeys">
+      <ae-modal
+          style="z-index: 100; text-align: center;"
+          v-if="modalVisible && !storedKeys"
+          @close="modalVisible = false"
+          title="Security"
+      >
+
+        <template v-if="hasPrivateKey">
+          <form id="pw-form" @submit.prevent="onUnlockWallet">
+            <ae-input placeholder="Type in your password to unlock your wallet" type="password" v-model="password"/>
+          </form>
+          <ae-button form="pw-form" type="dramatic">Unlock</ae-button>
+        </template>
+        <template v-else>
+          <ae-address v-if="publicKey" show-avatar size='short' :address="publicKey"></ae-address>Secure your wallet
+          <form id="secure-form" @submit.prevent="onSecureWallet">
+            <ae-input placeholder="Choose a password to select your wallet" type="password" v-model="password" />
+          </form>
+          <ae-button form="secure-form">Save</ae-button>
+        </template>
+      </ae-modal>
+    </template>
+    <template v-else>
+      <!--<ae-identity :identity="identity"></ae-identity>-->
+      <ae-address v-if="publicKey" show-avatar size='short' :address="publicKey"></ae-address>
+      <!--<faucet></faucet>-->
+    </template>
+  </div>
+</template>
+
+<script>
+  import {Crypto} from 'aepp-sdk'
+  import {mapGetters} from 'vuex'
+  import { AeButton, AeAddress, AeInput, AeIdentity, AeModal } from '@aeternity/aepp-components'
+  import Faucet from './Faucet'
+
+  export default {
+    name: 'Wallet',
+    components: {
+      Faucet,
+      AeButton,
+      AeAddress,
+      AeInput,
+      AeIdentity,
+      AeModal
+    },
+    data () {
+      return {
+        modalVisible: true,
+        password: '',
+        hasPrivateKey: false,
+        rawKey: undefined,
```

Comment:
Why set keys explicitly but assign undefined, which means the keys will effectively not be defined in the returned object..?